keps/sig-network/3766-referencegrant/README.md

```diff
@@ -0,0 +1,492 @@
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Other
+  - Describe the mechanism: Enable beta ReferenceGrant API
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+    No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+    No
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The API would become accessible again, implementing controllers may need to be
+restarted to pick up the presence of this API.
+
+###### Are there any tests for feature enablement/disablement?
+
+No
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+API enablement may not work, but that would not be unique to this API.
+
+###### What specific metrics should inform a rollback?
+
+N/A, this is just an API
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+N/A, this is just an API
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+N/A, this is just an API
+
+###### How can an operator determine if the feature is in use by workloads?
+```
+kubectl get referencegrants --all-namespaces
+```
+
+###### How can someone using this feature know that it is working for their instance?
+
+This will be dependent on the API that ReferenceGrant is used with. In Gateway API,
+each resource has clear status conditions that reflect the validity of a cross-namespace
+reference.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A, this is just an API
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+N/A, this is just an API
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+No
+
+### Dependencies
+
+N/A, this is just an API
+
+###### Does this feature depend on any specific services running in the cluster?
+
+- API Server
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+Yes, users may install controllers that watch for changes to ReferenceGrants.
+Users may also create ReferenceGrants to enable cross-namespace references.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+API Type: ReferenceGrant
+Supported Number of Objects per Cluster: No limit 
+Supported Number of Objects per Namespace: No limit
```

Comment:
not sure if there are global custom resource limits that sig-api-machinery or sig-scalability would prefer be added here